### ISSUE TITLE
Admin links redirect to WooCommerce admin home

### ISSUE THREAD

**Issue by gikaragia** [Automattic/wp-calypso#73547]
### Quick summary

We encountered this issue when testing Sensei's .com [onboarding flow](https://wordpress.com/setup/sensei/). The issue is happening once in 10-15 tries and we think that it is related to an inconsistency between the cache in an Atomic site and the actual options.

Although we encountered the issue in Sensei flow, this issue can happen to any newly transferred site to Atomic that has a new plugin installed. For example, you can also reproduce this if you try to install Sensei Pro on a site with business plan that isn't transferred yet. The issue seems to be happening for some time now, you can find more examples in this thread: p1676886444473859-slack-C7YPW6K40

The inconsistency in the cache can be seen by using the following commands:
```
wp> get_option('woocommerce_version');
=> bool(false)

wp db query "select * from wp_options where option_name = 'woocommerce_version'"
+-----------+---------------------+--------------+---------+
| option_id | option_name         | option_value | autoload |
+-----------+---------------------+--------------+----------+
|       621 | woocommerce_version | 7.4.0        | yes      |
+-----------+---------------------+--------------+----------+
```

If you run `wp cache flush` in the affected site the issue stops happening and the site starts to work normally. We are thinking to implement the following workaround for our case:
```
function sensei_fix_cache_incosistency() {
	$notoptions = wp_cache_get( 'notoptions', 'options' );

	if ( true === $notoptions[ 'woocommerce_version'] && is_plugin_active( 'woocommerce/woocommerce.php' ) ) {
		wp_cache_flush();
	}
}

add_action( 'woocommerce_installed', 'sensei_fix_cache_incosistency' );
```

Jetpack status seems to be ok:
```
wp jetpack status
Checking status for https://lasttest17.wpcomstaging.com
TEST RESULTS:
Passed:  test__blog_token_if_exists
Passed:  test__check_if_connected
Passed:  test__master_user_exists_on_site
Passed:  test__master_user_can_manage_options
Passed:  test__xml_parser_available
Passed:  test__outbound_http
Passed:  test__outbound_https
Passed:  test__identity_crisis
Passed:  test__connection_token_health
Passed:  test__wpcom_connection_test
Skipped: test__server_port_value
         The server port values are not defined. This is most common when running PHP via a CLI.
Skipped: test__full_sync_health
Passed:  test__sync_health
Success: Jetpack is currently connected to WordPress.com
```

### Steps to reproduce

1. Go to https://wordpress.com/setup/sensei/
2. Complete the steps to create a new site
3. *Important* When asked to 'Choose the purpose of your site' make sure that you select the 'Sell courses and generate income' option. This will install WooCommerce on the Atomic site.
4. Complete the flow and wait for the site to be created.
5. In the newly created site, click on Sensei -> Courses. Other links have the same issue e.g. Jetpack Dashboard, Woo Products etc.

### What you expected to happen

To go the the page that I clicked.

### What actually happened

I am redirected to the WooCommerce home. Moreover, whenever I click another link again I am always redirected to the WC home page.

![atomic-bug-wc-admin](https://user-images.githubusercontent.com/53191348/220163479-51453c3e-0774-416f-8e8b-250ad030a0f7.gif)

Sometimes, WC's onboarding flow is triggered too but when this happens the redirect is observed only once.

### Impact

Some (< 50%)

### Available workarounds?

No and the platform is unusable

### Platform (Simple and/or Atomic)

Atomic

**Comment by gikaragia** [comment]
This probably relates to https://github.com/Automattic/wp-calypso/issues/65214. Not sure if it should be treated as a separate issue.

**Comment by cuemarie** [comment]
📌 SCRUBBING : RESULT - *Replicated / Could Not Replicate / Uncertain*
- Tested on AT: Did not replicate (1 try)

📌 FINDINGS/SCREENSHOTS/VIDEO
- I followed the steps to replicate and was not able to see the issue; Sensei > Courses loaded the courses page right after the site was created.
- OP noted this issue happens in one out of every 10-15 tries:

> The issue is happening once in 10-15 tries and we think that it is related to an inconsistency between the cache in an Atomic site and the actual options.

Given the priority level and the long-standing issue https://github.com/Automattic/wp-calypso/issues/65214 , leaving this open for Quality Squad to try to dig into further.

📌 ACTIONS
- Marked as Triaged for Quality Squad review

**Comment by gikaragia** [comment]
An update on this issue.

We have changed the flow to not install WooCommerce so ATM it's not possible to reproduce it using the above steps.

There was some investigation done here: p9o2xV-2Pm-p2

The root cause of this issue seems to be a race condition and we have taken some steps to mitigate it, however I believe that the issue can happen again when WC is installed.
